<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Ethan Zhong, Toño Urbán" />


<title>Diffusion approximation of genetic drift</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">fiveMinuteStats</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/stephens999/fiveMinuteStats">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Diffusion approximation of genetic
drift</h1>
<h4 class="author">Ethan Zhong, Toño Urbán</h4>
<h4 class="date">3/17/2022</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-03-18
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong>
<code>~/Documents/Winter_Quarter_2022/Fundamentals/jaurbanChicago.github.io/bin/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed12345code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(12345)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed12345code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(12345)</code> was run prior to running the
code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomjaurbanChicagojaurbanChicagogithubiotree571343e27b9118d61ef613355b755f6cb8ab2e03targetblank571343ea">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/jaurbanChicago/jaurbanChicago.github.io/tree/571343e27b9118d61ef613355b755f6cb8ab2e03" target="_blank">571343e</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomjaurbanChicagojaurbanChicagogithubiotree571343e27b9118d61ef613355b755f6cb8ab2e03targetblank571343ea"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/jaurbanChicago/jaurbanChicago.github.io/tree/571343e27b9118d61ef613355b755f6cb8ab2e03" target="_blank">571343e</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Untracked files:
    Untracked:  .DS_Store
    Untracked:  .Rhistory

Unstaged changes:
    Deleted:    New Folder With Items/.DS_Store
    Deleted:    New Folder With Items/Genetic_Drift_Markov_Chain.Rmd
    Deleted:    New Folder With Items/_site.yml
    Deleted:    New Folder With Items/include/footer.html
    Modified:   bin/Genetic_Drift_Intro.Rmd
    Modified:   bin/Genetic_Drift_Markov.Rmd
    Modified:   bin/diffusion_approx.Rmd
    Modified:   bin/diffusion_process.Rmd
    Deleted:    index.html

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>bin/diffusion_approx.Rmd</code>) and HTML
(<code>docs/diffusion_approx.html</code>) files. If you’ve configured a
remote Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jaurbanChicago/jaurbanChicago.github.io/blob/742e5e63d0481f4d59459539a011273ff33a57c5/bin/diffusion_approx.Rmd" target="_blank">742e5e6</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-18
</td>
<td>
Added diffusion_approx
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jaurbanChicago/jaurbanChicago.github.io/742e5e63d0481f4d59459539a011273ff33a57c5/docs/diffusion_approx.html" target="_blank">742e5e6</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-18
</td>
<td>
Added diffusion_approx
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jaurbanChicago/jaurbanChicago.github.io/blob/644b273108b7cd2f24903218ab55b284ca4373c6/bin/diffusion_approx.Rmd" target="_blank">644b273</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-18
</td>
<td>
Added diffusion_approx.Rmd file
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>While the previous vignette provides a mathematically satisfactory
(without invoking advanced concepts beyond calculus) derivation of the
Kolmogorov forward/backward equation for time-homogenenous diffusion
process, it doesn’t convey too much intuition. In this vignette, we will
study the classic example of using diffusion process to approximate
genetic drift (Kimura 1955). As we will see, not only does it provide an
appealing model for continuous variations of allele frequency, it also
offers an intuitive justification of the Kolmogorov forward
equation.</p>
<div id="prerequisites" class="section level3">
<h3>Prerequisites</h3>
<ul>
<li><strong>A good understanding of the material in previous
vignettes</strong></li>
<li><strong>Able to reason continuous variation and infinitesimal
changes</strong></li>
</ul>
</div>
<div id="introduction" class="section level3">
<h3>Introduction</h3>
<p>The term <strong>diffusion</strong> refers to the general phenomenon
where particles move from a high concentration region to a lower
concentration region. In genetic drift, we use diffusion to describe the
“flattening” effect of random drift on the allele state distributions.
In 1956, Peter Buri used a <a
href="https://www.jstor.org/stable/2406998?seq=1">beautiful
experiment</a> to demonstrate this effect. Briefly, Buri began the
experiment with 107 populations of D. melanogaster (fruit fly). In each
population, the frequency of a target allele called <span
class="math inline">\(\text{bw}^{75}\)</span> is exactly 0.5. Buri
tracked the allele frequency changes over 19 generations by inferring
the genotype of each fly from its eye color. The changes in allele
frequency distribution vividly describe the effect of genetic drift on
many populations.</p>
<p><img src="drift_3d.png" width="50%" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-drift_3d">
Past versions of drift_3d.png
</button>
</p>
<div id="fig-drift_3d" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jaurbanChicago/jaurbanChicago.github.io/blob/742e5e63d0481f4d59459539a011273ff33a57c5/docs/drift_3d.png" target="_blank">742e5e6</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>In the <a
href="http://rpdata.caltech.edu/courses/bi1_2017/homework/hw4_pop_gen_Sp2017.pdf">picture
above</a>, all the masses are initially concentrated at the center. In a
few generations, the masses spread out fairly evenly. As time passes,
some alleles reach fixation and some are driven out of the
population.</p>
</div>
<div id="derivation-of-forward-equation" class="section level3">
<h3>Derivation of forward equation</h3>
<p>In fact, compared to our derivation, Kimura’s intuitive justification
applies to a broader class of diffusion processes, in that it doesn’t
assume time-homogeneity.<br />
<br />
Consider a locus with a pair of alleles <span
class="math inline">\(A_1\)</span> and <span
class="math inline">\(A_2\)</span> segregating in a population. Let
<span class="math inline">\(A_1\)</span> be the focal allele with
frequency <span class="math inline">\(p\)</span> at time 0. Following
our notation in the previous vignette, <span
class="math inline">\(\phi(p,0,x,t)\)</span> is the transition density
for the frequency of <span class="math inline">\(A_1\)</span>, <span
class="math inline">\(x\)</span>, at time <span
class="math inline">\(t\)</span>. Note that we use <span
class="math inline">\(\phi(\cdot)\)</span> instead of <span
class="math inline">\(p(\cdot)\)</span> to avoid confusion between
density and initial frequency. We can further simplify our notation to
<span class="math inline">\(\phi(x,t)\)</span> as 0 and <span
class="math inline">\(p\)</span> are both fixed quantities. The forward
equation is: <span class="math display">\[
\frac{d}{dt}\phi(x,t) = -\frac{d}{dx}\{\phi(x,t)\mu(x, t)\} +
\frac{1}{2}\frac{d^2}{dx^2}\{\phi(x,t)\sigma^2(x,t) \}
\]</span> Let’s pause for a second and think about the interpretation of
<span class="math inline">\(\phi(x,t)\)</span>. First of all, we
highlight the difference between diffusion process and the
characterization of genetic drift as binomial sampling (as demonstrated
in the first vignette). In the binomial sampling model, we focus on the
behavior in a single population. However, when we switch to Markov chain
and diffusion process, we are essentially modelling the <em>average
behavior</em> of allele frequency in many populations. Consider some
tiny <span class="math inline">\(dx &gt; 0\)</span>, then <span
class="math inline">\(\phi(x,t) dx\)</span> can be thought of as the
proportion of populations with allele frequency <span
class="math inline">\(x\)</span> among an infinite number of “identical”
Wright-Fisher populations. Sometimes we also call <span
class="math inline">\(\phi(x,t) dx\)</span> the frequency of class <span
class="math inline">\(x\)</span> at time <span
class="math inline">\(t\)</span>. In other words, think of a single
population as a small machine that runs its own drifting process. Now we
have many such population, each of which is a small machine that runs a
drifting process. Although the same set of “instructions” (parameters of
genetic drift) are sent to each machine, sampling variation in the
drifting process leads to allele frequency variation among populations.
If we pause the process at a particular time <span
class="math inline">\(t\)</span>, we obtain the allele frequency
distribution at time <span class="math inline">\(t\)</span> (as a
function of <span class="math inline">\(x\)</span>, the dummy variable
for allele frequency), which is exactly <span
class="math inline">\(\phi(x,t)\)</span>. Kimura (1955) has a nice
diagram that visualizes <span
class="math inline">\(\phi(x,t)\)</span>.</p>
<p><img src="kimura1955.png" width="30%" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-kimura1955">
Past versions of kimura1955.png
</button>
</p>
<div id="fig-kimura1955" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jaurbanChicago/jaurbanChicago.github.io/blob/742e5e63d0481f4d59459539a011273ff33a57c5/docs/kimura1955.png" target="_blank">742e5e6</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The distribution in approximated by bins of width <span
class="math inline">\(h\)</span>. The approximation gets better as <span
class="math inline">\(h\)</span> becomes smaller, so think of <span
class="math inline">\(h\)</span> tiny. Because the density in each bin
is constant, there are only three classes here, represented by the
middle points <span class="math inline">\(x-h\)</span>, <span
class="math inline">\(x\)</span>, and <span
class="math inline">\(x+h\)</span>, respectively. Consider population of
allele frequency <span class="math inline">\(x\)</span> (henthforce
class <span class="math inline">\(x\)</span> and similarly for other
classes), we want to describe its probability of moving to another class
after some time <span class="math inline">\(\Delta t\)</span>. Recall in
the definition of diffusion process, there is a technical condition
which ensures continuous path. Therefore, if we take <span
class="math inline">\(\Delta t\)</span> sufficiently small, it will move
to either class <span class="math inline">\(x-h\)</span> or class <span
class="math inline">\(x+h\)</span>.<br />
<br />
We also consider two forces that drive the population to different
classes: systematic pressure and randomness. Let <span
class="math inline">\(m(x,t)\Delta t\)</span> be the probability that
the population moves to class <span class="math inline">\(x+h\)</span>
after time <span class="math inline">\(\Delta t\)</span> due to
systematic pressure. Note that the pressure can be specified in either
direction without loss of generality. Let <span
class="math inline">\(v(x,t)\Delta t\)</span> be the probability that
the population move to a different class after time <span
class="math inline">\(\Delta t\)</span> due to randomness. Therefore, if
the population leaves the current class (say, class <span
class="math inline">\(x\)</span>) by randomness, half of time it moves
to class <span class="math inline">\(x-h\)</span> and half of the time
to class <span class="math inline">\(x+h\)</span>. We can write the
density after time <span class="math inline">\(\Delta t\)</span> as:
<span class="math display">\[
\begin{aligned}
\phi(x,t+\Delta t)h &amp;= \phi(x,t)h \\ &amp;- [v(x,t)+m(x,t)]\Delta t
\phi(x,t)h \hspace{10mm} (\star) \\ &amp;+ \frac{1}{2}v(x-h,t)\Delta t
\phi(x-h,t) + \frac{1}{2}v(x+h, t)\Delta t \phi(x+h,t)h \hspace{10mm}
(\star\star)\\ &amp;+ m(x-h,t)\Delta t \phi(x-h, t)h \hspace{10mm}
(\star\star\star)
\end{aligned}
\]</span> Let’s translate the above equation into plain English. <span
class="math inline">\((\star)\)</span> represents the probability mass
leaving class <span class="math inline">\(x\)</span> per <span
class="math inline">\(\Delta t\)</span> at time <span
class="math inline">\(t\)</span>; <span
class="math inline">\((\star\star)\)</span> represents the probability
mass moving into class <span class="math inline">\(x\)</span> per <span
class="math inline">\(\Delta t\)</span> from the other two classes due
to randomness; <span class="math inline">\((\star\star\star)\)</span>
represents the probability mass moving into class <span
class="math inline">\(x\)</span> per <span class="math inline">\(\Delta
t\)</span> from class <span class="math inline">\(x-h\)</span> by
systematic pressure (i.e.: the wind only blows from left to right).
Together, <span class="math inline">\((\star\star)\)</span> and <span
class="math inline">\((\star\star\star)\)</span> represent the
probability mass entering class <span class="math inline">\(x\)</span>
per <span class="math inline">\(\Delta t\)</span> at time <span
class="math inline">\(t\)</span>. Hence, the previous equation says:
<span class="math display">\[
\begin{aligned}
\text{the probability mass in class } x \text{ at time } (t + \Delta t)
&amp;= \text{the probability mass in class } x \text{ at time } t \\
&amp;- \text{the probability mass leaving class } x \text{ per } \Delta
t \text{ at time } t \\ &amp;+ \text{the probability mass entering class
} x \text{ per } \Delta t \text{ at time } t
\end{aligned}
\]</span> Cool! Now we can define <span
class="math inline">\(\sigma^2(x,t)\)</span> and <span
class="math inline">\(\mu(x,t)\)</span>. Let: <span
class="math display">\[
\begin{aligned}
\sigma^2(x,t) \Delta t &amp;= h^2 \frac{1}{2}v(x,t)\Delta t + (-h)^2
\frac{1}{2}v(x,t)\Delta t\\
\mu(x,t) \Delta t &amp;= hm(x,t)\Delta t
\end{aligned}
\]</span> In other words, <span class="math inline">\(\sigma^2(x,t)
\Delta t\)</span> is the variance of the per <span
class="math inline">\(\Delta t\)</span> change in <span
class="math inline">\(x\)</span> due to randomness, and <span
class="math inline">\(\mu(x,t) \Delta t\)</span> is the average per
<span class="math inline">\(\Delta t\)</span> change in <span
class="math inline">\(x\)</span>. Dividing both sides by <span
class="math inline">\(\Delta t\)</span>, we have: <span
class="math display">\[
\begin{aligned}
\sigma^2(x,t) &amp;= h^2 v(x,t)\\
\mu(x,t) &amp;= hm(x,t)
\end{aligned}
\]</span> Although <span class="math inline">\(\Delta t\)</span> is
short, it still counts as a segment. Now we are looking at a particle.
We should interpret <span class="math inline">\(\sigma^2(x,t)\)</span>
as the variance per infinitesimal time change in <span
class="math inline">\(x\)</span> due to randomness and <span
class="math inline">\(\mu(x,t)\)</span> the average per infinitesimal
time change in <span class="math inline">\(x\)</span>. Recall in the
definition of diffusion process, we call <span
class="math inline">\(\mu(x,t)\)</span> the <strong>infinitesimal
mean</strong> and <span class="math inline">\(\sigma^2(x,t)\)</span> the
<strong>infinitesimal variance</strong>.<br />
<br />
Eventually, we want to find an expression for the time derivative: <span
class="math display">\[
\lim_{\Delta t \rightarrow 0}  \frac{\phi(x,t+\Delta t) -
\phi(x,t)}{\Delta t}
\]</span> This is easy because we can substitute <span
class="math inline">\(\sigma^2(x,t)\)</span> and <span
class="math inline">\(\mu(x,t)\)</span> back to the equation for <span
class="math inline">\(\phi(x,t+\Delta t)h\)</span>. With some standard
algebraic manipulations, we have the desired forward equation (remember
to take the limit as <span class="math inline">\(h \rightarrow
0\)</span> because we start with an approximation of the distribution):
<span class="math display">\[
\frac{d}{dt}\phi(x,t) = -\frac{d}{dx}\{\phi(x,t)\mu(x, t)\} +
\frac{1}{2}\frac{d^2}{dx^2}\{\phi(x,t)\sigma^2(x,t) \}
\]</span></p>
</div>
<div id="main-reference" class="section level3">
<h3>Main reference</h3>
<ul>
<li><em>Stochastic processes and distribution of gene frequencies under
natural selection by Motoo Kimura (1955)</em></li>
</ul>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.2 (2021-11-01)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.8.2     rstudioapi_0.13  whisker_0.4      knitr_1.37      
 [5] magrittr_2.0.2   workflowr_1.7.0  R6_2.5.1         rlang_1.0.2     
 [9] fastmap_1.1.0    fansi_1.0.2      highr_0.9        stringr_1.4.0   
[13] tools_4.1.2      xfun_0.29        utf8_1.2.2       cli_3.2.0       
[17] jquerylib_0.1.4  git2r_0.29.0     htmltools_0.5.2  ellipsis_0.3.2  
[21] yaml_2.2.2       digest_0.6.29    rprojroot_2.0.2  lifecycle_1.0.1 
[25] tibble_3.1.6     crayon_1.5.0     later_1.3.0      vctrs_0.3.8     
[29] sass_0.4.0       promises_1.2.0.1 fs_1.5.2         glue_1.6.2      
[33] evaluate_0.14    rmarkdown_2.11   stringi_1.7.6    pillar_1.7.0    
[37] compiler_4.1.2   bslib_0.3.1      jsonlite_1.8.0   httpuv_1.6.5    
[41] pkgconfig_2.0.3 </code></pre>
</div>
</div>

<hr>
<p>
    This site was created with <a href="http://rmarkdown.rstudio.com">R Markdown</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
