<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jose Antonio Urban Aragon, Ethan Zhong" />


<title>Genetic Drift in a Finite Population as an Example of Stochastic Modeling</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">fiveMinuteStats</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/stephens999/fiveMinuteStats">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Genetic Drift in a Finite Population as an
Example of Stochastic Modeling</h1>
<h4 class="author">Jose Antonio Urban Aragon, Ethan Zhong</h4>
<h4 class="date">3/13/2022</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-03-17
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong>
<code>~/Documents/Winter_Quarter_2022/Fundamentals/jaurbanChicago.github.io/bin/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed12345code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(12345)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed12345code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(12345)</code> was run prior to running the
code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomjaurbanChicagojaurbanChicagogithubiotreedfec2ffef0638254ab6743b43a93bf4d4a4f7368targetblankdfec2ffa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/jaurbanChicago/jaurbanChicago.github.io/tree/dfec2ffef0638254ab6743b43a93bf4d4a4f7368" target="_blank">dfec2ff</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomjaurbanChicagojaurbanChicagogithubiotreedfec2ffef0638254ab6743b43a93bf4d4a4f7368targetblankdfec2ffa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/jaurbanChicago/jaurbanChicago.github.io/tree/dfec2ffef0638254ab6743b43a93bf4d4a4f7368" target="_blank">dfec2ff</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Untracked files:
    Untracked:  .DS_Store
    Untracked:  .Rhistory

Unstaged changes:
    Deleted:    New Folder With Items/.DS_Store
    Deleted:    New Folder With Items/Genetic_Drift_Markov_Chain.Rmd
    Deleted:    New Folder With Items/_site.yml
    Deleted:    New Folder With Items/include/footer.html
    Modified:   bin/Genetic_Drift_Introduction.Rmd
    Deleted:    index.html

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>bin/Genetic_Drift_Introduction.Rmd</code>)
and HTML (<code>docs/Genetic_Drift_Introduction.html</code>) files. If
you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jaurbanChicago/jaurbanChicago.github.io/blob/dfec2ffef0638254ab6743b43a93bf4d4a4f7368/bin/Genetic_Drift_Introduction.Rmd" target="_blank">dfec2ff</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-17
</td>
<td>
Corrected intro html and rmd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jaurbanChicago/jaurbanChicago.github.io/dfec2ffef0638254ab6743b43a93bf4d4a4f7368/docs/Genetic_Drift_Introduction.html" target="_blank">dfec2ff</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-17
</td>
<td>
Corrected intro html and rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jaurbanChicago/jaurbanChicago.github.io/blob/06472ac8b74c387e131c89f4d721052fd7c868e1/bin/Genetic_Drift_Introduction.Rmd" target="_blank">06472ac</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-17
</td>
<td>
Added diffusion_process rmd and html
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jaurbanChicago/jaurbanChicago.github.io/06472ac8b74c387e131c89f4d721052fd7c868e1/docs/Genetic_Drift_Introduction.html" target="_blank">06472ac</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-17
</td>
<td>
Added diffusion_process rmd and html
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jaurbanChicago/jaurbanChicago.github.io/blob/7472e3f4eae1bee1c7de56afcddcf79202135a35/bin/Genetic_Drift_Introduction.Rmd" target="_blank">7472e3f</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-17
</td>
<td>
Updates both rmd and html
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jaurbanChicago/jaurbanChicago.github.io/7472e3f4eae1bee1c7de56afcddcf79202135a35/docs/Genetic_Drift_Introduction.html" target="_blank">7472e3f</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-17
</td>
<td>
Updates both rmd and html
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jaurbanChicago/jaurbanChicago.github.io/blob/0d8c343b4a9da2b48bf76780b3460ddd808a7d9e/bin/Genetic_Drift_Introduction.Rmd" target="_blank">0d8c343</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-17
</td>
<td>
Updates both rmd and html
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jaurbanChicago/jaurbanChicago.github.io/0d8c343b4a9da2b48bf76780b3460ddd808a7d9e/docs/Genetic_Drift_Introduction.html" target="_blank">0d8c343</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-17
</td>
<td>
Updates both rmd and html
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jaurbanChicago/jaurbanChicago.github.io/blob/6ad5355b0d2e4a20eb74b4b4e3e3a8870bb591b2/bin/Genetic_Drift_Introduction.Rmd" target="_blank">6ad5355</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-17
</td>
<td>
Minor updates
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jaurbanChicago/jaurbanChicago.github.io/6ad5355b0d2e4a20eb74b4b4e3e3a8870bb591b2/docs/Genetic_Drift_Introduction.html" target="_blank">6ad5355</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-17
</td>
<td>
Minor updates
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jaurbanChicago/jaurbanChicago.github.io/blob/b4aee82cc2a2bc82b00f509d7c399937b39cc6e1/bin/Genetic_Drift_Introduction.Rmd" target="_blank">b4aee82</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-17
</td>
<td>
Updated rmd and html files
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jaurbanChicago/jaurbanChicago.github.io/b4aee82cc2a2bc82b00f509d7c399937b39cc6e1/docs/Genetic_Drift_Introduction.html" target="_blank">b4aee82</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-17
</td>
<td>
Updated rmd and html files
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jaurbanChicago/jaurbanChicago.github.io/blob/de32f0f7da3a84d1f7b7a1fc1f504f96470ad100/bin/Genetic_Drift_Introduction.Rmd" target="_blank">de32f0f</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-16
</td>
<td>
Updated intro rmd and html
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jaurbanChicago/jaurbanChicago.github.io/de32f0f7da3a84d1f7b7a1fc1f504f96470ad100/docs/Genetic_Drift_Introduction.html" target="_blank">de32f0f</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-16
</td>
<td>
Updated intro rmd and html
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jaurbanChicago/jaurbanChicago.github.io/blob/5ddeea6940e245025109be218a750ac0eebd5e81/bin/Genetic_Drift_Introduction.Rmd" target="_blank">5ddeea6</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-16
</td>
<td>
Updated intro rmd and html and added Markov model rmd and html
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jaurbanChicago/jaurbanChicago.github.io/5ddeea6940e245025109be218a750ac0eebd5e81/docs/Genetic_Drift_Introduction.html" target="_blank">5ddeea6</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-16
</td>
<td>
Updated intro rmd and html and added Markov model rmd and html
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jaurbanChicago/jaurbanChicago.github.io/blob/560a9f1c63551ccc061cce6abf4a6af3d0d21de8/bin/Genetic_Drift_Introduction.Rmd" target="_blank">560a9f1</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-16
</td>
<td>
Updated intro rmd and html
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jaurbanChicago/jaurbanChicago.github.io/560a9f1c63551ccc061cce6abf4a6af3d0d21de8/docs/Genetic_Drift_Introduction.html" target="_blank">560a9f1</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-16
</td>
<td>
Updated intro rmd and html
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jaurbanChicago/jaurbanChicago.github.io/blob/62ed45081892587635a624b7e6e3de969ae85b35/bin/Genetic_Drift_Introduction.Rmd" target="_blank">62ed450</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-15
</td>
<td>
Updated intro rmd and html
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jaurbanChicago/jaurbanChicago.github.io/62ed45081892587635a624b7e6e3de969ae85b35/docs/Genetic_Drift_Introduction.html" target="_blank">62ed450</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-15
</td>
<td>
Updated intro rmd and html
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jaurbanChicago/jaurbanChicago.github.io/blob/e705087b630ae60fb18b417ea5ed3d5e7034ccc8/bin/Genetic_Drift_Introduction.Rmd" target="_blank">e705087</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-15
</td>
<td>
Updated intro rmd and html
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jaurbanChicago/jaurbanChicago.github.io/e705087b630ae60fb18b417ea5ed3d5e7034ccc8/docs/Genetic_Drift_Introduction.html" target="_blank">e705087</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-15
</td>
<td>
Updated intro rmd and html
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jaurbanChicago/jaurbanChicago.github.io/blob/4ba7a1189a531a6d1b55038a3336f250f0e79f0a/bin/Genetic_Drift_Introduction.Rmd" target="_blank">4ba7a11</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-14
</td>
<td>
Updated intro rmd and html
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jaurbanChicago/jaurbanChicago.github.io/4ba7a1189a531a6d1b55038a3336f250f0e79f0a/docs/Genetic_Drift_Introduction.html" target="_blank">4ba7a11</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-14
</td>
<td>
Updated intro rmd and html
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jaurbanChicago/jaurbanChicago.github.io/aa49b2d1472d2ee647845d82643813433c6600b4/docs/Genetic_Drift_Introduction.html" target="_blank">aa49b2d</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-14
</td>
<td>
Updated intro html
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jaurbanChicago/jaurbanChicago.github.io/blob/1e475f283a5b10ade3a7872ad6a368df76505311/bin/Genetic_Drift_Introduction.Rmd" target="_blank">1e475f2</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-14
</td>
<td>
Updated intro rmd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jaurbanChicago/jaurbanChicago.github.io/5d7569c7ba16dd3d7779d69d558a1c74c4f481f0/docs/Genetic_Drift_Introduction.html" target="_blank">5d7569c</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-14
</td>
<td>
Added Genetic_Drift_Introduction.html
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jaurbanChicago/jaurbanChicago.github.io/blob/7be8910f6538e8dbb5702e76ee5c18f02079ba93/bin/Genetic_Drift_Introduction.Rmd" target="_blank">7be8910</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-14
</td>
<td>
Added Genetic_Drift_Introduction.Rmd
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="pre-requisites" class="section level1">
<h1>Pre-Requisites</h1>
<ul>
<li><a
href="https://stephens999.github.io/fiveMinuteStats/wright_fisher_model.html">Wright-Fisher
Model</a></li>
<li><a
href="http://stephens999.github.io/fiveMinuteStats/markov_chains_discrete_intro.html">Discrete
Markov Chains</a></li>
<li>Basic notions of probability theory</li>
<li>Basic notions of population genetics</li>
</ul>
</div>
<div id="genetic-drift-in-finite-populations" class="section level1">
<h1>Genetic Drift in Finite Populations</h1>
<p>Much population genetic theory concentrates on describing the changes
of allele frequencies through time. The two most important factors
influencing allele trajectories through time are natural selection and
<strong>genetic drift</strong> [1]. In these vignettes, we will focus on
genetic drift and how this process may be modeled as a <a
href="https://en.wikipedia.org/wiki/Stochastic_process">stochastic
process</a> in a <a
href="https://stephens999.github.io/fiveMinuteStats/wright_fisher_model.html">Wright-Fisher
population</a>.</p>
<p>Genetic drift is the random change of allele frequencies through time
in populations of finite size [1,2]. Different random processes could
affect the outcome of genetic drift, and they might have nothing to do
with the individuals’ fitness [1]. For example, some individuals may die
before they reach reproductive age because of extrinsic effects not
related to their genetic makeup [1]. Additionally, there is inherent
randomness in Mendelian segregation, and these effects could become more
apparent when populations are small (we will shortly see why this is the
case) [1,2].</p>
</div>
<div id="brief-introduction-to-a-wright-fisher-model"
class="section level1">
<h1>Brief Introduction to a Wright-Fisher Model</h1>
<p>Before delving into the specific probabilistic models of genetic
drift, it is important to summarize the main basic assumptions of the
Wright-Fisher population. The Wright-Fisher population is an idealized
biological scenario where the models of genetic drift are useful to
describe allele frequencies changes due to random sampling error [1].
Although the Wright-Fisher model is biologically unrealistic, it has
been extremely useful to model genetic drift (and other evolutionary
processes) in a relatively simple fashion [1]. Some of the basic
assumptions of the Wright-Fisher model include [1]:</p>
<ul>
<li>discrete, non-overlapping generations</li>
<li>numbers of females and males are equal</li>
<li>size of population (<span class="math inline">\(N\)</span>
individuals ) remains constant through time and is finite</li>
<li>no natural selection (all individuals are equal in their production
of gametes and all gametes are equally viable)</li>
<li>infinite pool of gametes</li>
<li>finite sample of <span class="math inline">\(2N\)</span> alleles is
drawn from the gamete pool to form the next generation of <span
class="math inline">\(N\)</span> individuals</li>
<li>the sampling errors that originate genetic drift occur when the
<span class="math inline">\(2N\)</span> alleles are drawn from the
population´s gamete pool</li>
</ul>
<center>
<p><img src="Figure_1.png" /></p>
Figure 1. <strong>A.</strong> Schematic of the Wright-Fisher model.
Sampling error -and genetic drift-, happens in the random draw of
gametes from generation <span class="math inline">\(t\)</span> to
generation <span class="math inline">\(t+1\)</span>. <em>Image from
Hamilton, M.B. (2008)</em> <strong>B.</strong> Sewall Wright (undated)
with a guinea pig as a professor at the University of Chicago.
<em>University of Chicago Photographic Archive, [apf digital item
number, e.g., apf12345], Hanna Holborn Gray Special Collections Research
Center, University of Chicago Library.</em> <strong>C.</strong> Ronald
Fisher at his calculator in 1958 <em>(Fisher Memorial Trust)</em>.
</center>
</div>
<div
id="models-used-to-describe-genetic-drift-no-mutation-andor-selection"
class="section level1">
<h1>Models Used to Describe Genetic Drift (No Mutation and/or
Selection)</h1>
<p>In this section of the vignette, we will introduce three basic models
that can be used to analyze properties of genetic drift:</p>
<ul>
<li>Binomial distribution<br />
</li>
<li>Markov chain</li>
<li>Diffusion process (continuous time approximation to the Markov
chain)</li>
</ul>
<div id="binomial-distribution" class="section level3">
<h3>Binomial Distribution</h3>
<p>To understand genetic drift of a single two-allele locus in a
Wright-Fisher population, we must first get into the details of an
important discrete probability distribution: <strong>the binomial
distribution.</strong><br />
A binomial distribution is a discrete probability distribution that
defines the probability distribution for the sum of <span
class="math inline">\(N\)</span> independent samples of a Bernoulli
variable [1]. In terms of two alleles <span
class="math inline">\(A\)</span> and <span
class="math inline">\(a\)</span> and to describe the sampling of alleles
from a gamete pool of a N individuals population, the following formula
is useful:</p>
<p><span class="math display">\[
P_{i=A}={2N \choose i}p^{i}(1-p)^{2N-i} \\
\]</span> This formula describes the probability of sampling <span
class="math inline">\(i\)</span> <span class="math inline">\(A\)</span>
alleles in a sample of <span class="math inline">\(2N\)</span> alleles
from a population where the <span class="math inline">\(A\)</span>
allele has a frequency of <span class="math inline">\(p\)</span> and the
alternate <span class="math inline">\(a\)</span> allele has a frequency
of <span class="math inline">\(1-p\)</span>. The <span
class="math inline">\(p^i\)</span> and <span
class="math inline">\((1-p)^{2N-i}\)</span> terms estimate the
probability of observing a <span class="math inline">\(i\)</span> and
<span class="math inline">\(2N-i\)</span> independent events each with
probability <span class="math inline">\(p\)</span> and <span
class="math inline">\(1-p\)</span>, respectively [1]. With some
knowledge of the properties of the binomial probability distribution, we
can realize that the Wright-Fisher model is a very good approximation to
model genetic drift [1]. Some of the properties of the Bernoulli random
variables will helps to better understand some specific behaviors of
allele frequencies under the action of genetic drift. Some of these
properties are:</p>
<p><span class="math display">\[
\sigma^2=p(1-p) \space \mathbb{for \space a \space single \space
Bernoulli \space variable} \\
\mathbb{SE(standard \space error)}=\sqrt{\frac{2p(1-p)}{2N}} \\
\mathbb{Additionally, we \space can \space also \space write \space
formulas \space in \space terms \space of \space conditional \space
expectation \space and \space conditional \space variance:} \\
\mathbb{E}(p_{t+1}|p_t)=p_t \\
\mathbb{Var}(p_{t+1}|p_t)=\frac{p_t(1-p_t)}{2N} \\
p_{t}= \mathbb{frequency \space allele \space A \space in \space
generation \space t} \\
p_{t+1}= \mathbb{frequency \space allele \space A \space in \space
generation \space t+1}\\
\]</span></p>
<p>As we may infer from the formulas, the maximum variability in the
outcomes of the alleles’ binomial sampling process will occur when
<strong>the frequency <span class="math inline">\(p\)</span> of <span
class="math inline">\(A\)</span> equals the frequency <span
class="math inline">\(1-p\)</span> of <span
class="math inline">\(A\)</span> (<span
class="math inline">\(p=1-p=0.5\)</span>)</strong>[1]. This makes
intuitive sense because as the frequency of one of the alleles reaches
fixation , the sampling error that results in greatly increased
frequencies of the low-frequency allele in one generation will be very
highly unlikely [1]. On the other hand, when both alleles are equally
frequent or almost equally frequent, sampling error is equally likely to
increase or decrease allele frequency and its outcomes could vary more
widely along the spectrum of possible allele frequencies.</p>
<p>The following code will demonstrate how the variability of the
outcomes of the sampling process at time <span
class="math inline">\(t+1\)</span> depends,in part, to allele
frequencies at time <span class="math inline">\(t\)</span>:</p>
<pre class="r"><code>library(tidyverse)
# Vector of allele frequency for allele A
p &lt;-  seq(0,1,by=0.01) # Frequency p of A allele
N &lt;- 100 # Population of N individuals
# vector to store standard errors


fun_p &lt;- function(p,N){ # function to estimate the standard errors for each frequencies
  se &lt;- c()
  for(i in 1:length(p)){
    se[i] &lt;- sqrt((p[i]*(1-p[i]))/(2*N))
  }
  return(se)
}


SE &lt;- fun_p(p=p,N=N)

df &lt;- cbind(p,SE)
df &lt;- as.data.frame(df,col.names=names(&quot;p&quot;,&quot;SE&quot;))

# Plot of Allele Frequency p vs Standard Error
ggplot(df,aes(x=p,y=SE))+geom_line()+xlab(&quot;Allele Frequency p&quot;)+ylab(&quot;SE&quot;)+ggtitle(&quot;Allele Frequency vs Standard Error&quot;)+theme(plot.title = element_text(hjust = 0.5))+geom_hline(yintercept = max(SE),color=&quot;red&quot;)</code></pre>
<p><img src="figure/Genetic_Drift_Introduction.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-1">
Past versions of unnamed-chunk-1-1.png
</button>
</p>
<div id="fig-unnamed-chunk-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jaurbanChicago/jaurbanChicago.github.io/blob/560a9f1c63551ccc061cce6abf4a6af3d0d21de8/docs/figure/Genetic_Drift_Introduction.Rmd/unnamed-chunk-1-1.png" target="_blank">560a9f1</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#The horizontal red line determines the maximum value of SE.</code></pre>
<p>The above plot confirms what we just discussed. The maximum
variability of allele frequency changes happens when both alleles are
very close in frequency[1].</p>
<p>Another feature affecting the variability of binomial sampling errors
in genetic drift is the <strong>population size</strong>. As population
size increases, the variability of allele frequencies becomes less and
less, constraining the spectrum of possible changes in allele
frequencies moving forward in time and reducing the variance of the
sampling error process [1,2].</p>
<p>The following code will illustrate this:</p>
<pre class="r"><code>p &lt;- 0.5 # Set the p frequency to 0.5

N &lt;-seq(10,100,by=10) # Create the N vector

fun_p &lt;- function(p,N){ # function to estimate the standard errors for each population size
  se &lt;- c()
  for(i in 1:length(N)){
    se[i] &lt;- sqrt((p*(1-p))/(2*N[i]))
  }
  return(se)
}

SE &lt;- fun_p(p=p,N=N)

df &lt;- cbind(N, SE)
df &lt;- as.data.frame(df,col.names=names(&quot;N&quot;,&quot;SE&quot;))

# Plot of Allele Frequency p vs Standard Error
ggplot(df,aes(x=N,y=SE))+geom_line()+xlab(&quot;Population Size (N)&quot;)+ylab(&quot;SE at p=0.5&quot;)+ggtitle(&quot;Population Size vs Standard Error at p=0.5&quot;)+theme(plot.title = element_text(hjust = 0.5))+geom_hline(yintercept = max(SE),color=&quot;red&quot;)</code></pre>
<p><img src="figure/Genetic_Drift_Introduction.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jaurbanChicago/jaurbanChicago.github.io/blob/560a9f1c63551ccc061cce6abf4a6af3d0d21de8/docs/figure/Genetic_Drift_Introduction.Rmd/unnamed-chunk-2-1.png" target="_blank">560a9f1</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#The horizontal red line determines the maximum value of SE at p=0.5.</code></pre>
<p>Both features (population size and allele frequencies at time <span
class="math inline">\(t\)</span>) will be of paramount importance to
determine the outcomes of genetic drift in a given Wright-Fisher
population.</p>
</div>
<div id="markov-chain" class="section level3">
<h3>Markov Chain</h3>
<p>The Markov chain model of genetic drift will be explained in the
following <a
href="https://jaurbanchicago.github.io/docs/Genetic_Drift_Markov.html">vignette</a>.</p>
</div>
<div
id="diffusion-process-continuous-time-approximation-of-the-markov-chain-model"
class="section level3">
<h3>Diffusion Process (Continuous time approximation of the Markov chain
model)</h3>
<p>The diffusion process model of genetic drift will be explained in the
following vignettes:</p>
<ul>
<li><a
href="https://jaurbanchicago.github.io/docs/diffusion_process.html">Introduction
to Difussion Processes</a><br />
</li>
<li><a href="">Difussion Processes and Genetic Drift</a></li>
</ul>
</div>
</div>
<div id="references" class="section level1">
<h1>References</h1>
<ol style="list-style-type: decimal">
<li><p>Hamilton, M. B. (2008). <em>Population genetics</em>. John Wiley
&amp; Sons, 53-73.</p></li>
<li><p>Hartl, D. L. (2020).<em>A primer of population genetics and
genomics</em>. Oxford University Press,147-173.</p></li>
</ol>
<p>sessionInfo()</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.2 (2021-11-01)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] forcats_0.5.1   stringr_1.4.0   dplyr_1.0.8     purrr_0.3.4    
[5] readr_2.1.2     tidyr_1.2.0     tibble_3.1.6    ggplot2_3.3.5  
[9] tidyverse_1.3.1

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.8.2     lubridate_1.8.0  assertthat_0.2.1 rprojroot_2.0.2 
 [5] digest_0.6.29    utf8_1.2.2       R6_2.5.1         cellranger_1.1.0
 [9] backports_1.4.1  reprex_2.0.1     evaluate_0.14    highr_0.9       
[13] httr_1.4.2       pillar_1.7.0     rlang_1.0.2      readxl_1.3.1    
[17] rstudioapi_0.13  whisker_0.4      jquerylib_0.1.4  rmarkdown_2.11  
[21] labeling_0.4.2   munsell_0.5.0    broom_0.7.12     compiler_4.1.2  
[25] httpuv_1.6.5     modelr_0.1.8     xfun_0.29        pkgconfig_2.0.3 
[29] htmltools_0.5.2  tidyselect_1.1.2 workflowr_1.7.0  fansi_1.0.2     
[33] crayon_1.5.0     tzdb_0.2.0       dbplyr_2.1.1     withr_2.5.0     
[37] later_1.3.0      grid_4.1.2       jsonlite_1.8.0   gtable_0.3.0    
[41] lifecycle_1.0.1  DBI_1.1.2        git2r_0.29.0     magrittr_2.0.2  
[45] scales_1.1.1     cli_3.2.0        stringi_1.7.6    farver_2.1.0    
[49] fs_1.5.2         promises_1.2.0.1 xml2_1.3.3       bslib_0.3.1     
[53] ellipsis_0.3.2   generics_0.1.2   vctrs_0.3.8      tools_4.1.2     
[57] glue_1.6.2       hms_1.1.1        fastmap_1.1.0    yaml_2.2.2      
[61] colorspace_2.0-3 rvest_1.0.2      knitr_1.37       haven_2.4.3     
[65] sass_0.4.0      </code></pre>
</div>
</div>

<hr>
<p>
    This site was created with <a href="http://rmarkdown.rstudio.com">R Markdown</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
