<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jose Antonio Urban Aragon, Ethan Zhong" />


<title>Markov Chain Model of Genetic Drift</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">fiveMinuteStats</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/stephens999/fiveMinuteStats">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Markov Chain Model of Genetic Drift</h1>
<h4 class="author">Jose Antonio Urban Aragon, Ethan Zhong</h4>
<h4 class="date">3/16/2022</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-03-18
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong>
<code>~/Documents/Winter_Quarter_2022/Fundamentals/jaurbanChicago.github.io/bin/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed12345code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(12345)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed12345code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(12345)</code> was run prior to running the
code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomjaurbanChicagojaurbanChicagogithubiotree571343e27b9118d61ef613355b755f6cb8ab2e03targetblank571343ea">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/jaurbanChicago/jaurbanChicago.github.io/tree/571343e27b9118d61ef613355b755f6cb8ab2e03" target="_blank">571343e</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomjaurbanChicagojaurbanChicagogithubiotree571343e27b9118d61ef613355b755f6cb8ab2e03targetblank571343ea"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/jaurbanChicago/jaurbanChicago.github.io/tree/571343e27b9118d61ef613355b755f6cb8ab2e03" target="_blank">571343e</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Untracked files:
    Untracked:  .DS_Store
    Untracked:  .Rhistory

Unstaged changes:
    Deleted:    New Folder With Items/.DS_Store
    Deleted:    New Folder With Items/Genetic_Drift_Markov_Chain.Rmd
    Deleted:    New Folder With Items/_site.yml
    Deleted:    New Folder With Items/include/footer.html
    Modified:   bin/Genetic_Drift_Intro.Rmd
    Modified:   bin/Genetic_Drift_Markov.Rmd
    Deleted:    index.html

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>bin/Genetic_Drift_Markov.Rmd</code>) and
HTML (<code>docs/Genetic_Drift_Markov.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jaurbanChicago/jaurbanChicago.github.io/blob/fccedde7c6b18b10abe2d9f022ebee0283e2bad5/bin/Genetic_Drift_Markov.Rmd" target="_blank">fccedde</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-18
</td>
<td>
Updated markov
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jaurbanChicago/jaurbanChicago.github.io/fccedde7c6b18b10abe2d9f022ebee0283e2bad5/docs/Genetic_Drift_Markov.html" target="_blank">fccedde</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-18
</td>
<td>
Updated markov
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jaurbanChicago/jaurbanChicago.github.io/blob/af0fd73dae829ecb4fd82fd2da942c2863737c59/bin/Genetic_Drift_Markov.Rmd" target="_blank">af0fd73</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-18
</td>
<td>
markov html
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jaurbanChicago/jaurbanChicago.github.io/883962fbc60b397adbfde841d8136a554f07a6be/docs/Genetic_Drift_Markov.html" target="_blank">883962f</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-18
</td>
<td>
Updated markov html
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jaurbanChicago/jaurbanChicago.github.io/943c0b9a59eb99a982c58d82eedb21e32d7e7766/docs/Genetic_Drift_Markov.html" target="_blank">943c0b9</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-18
</td>
<td>
Updated markov html
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jaurbanChicago/jaurbanChicago.github.io/blob/b65811af4f60ffc1f71671089d08b420968a3d4a/bin/Genetic_Drift_Markov.Rmd" target="_blank">b65811a</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-18
</td>
<td>
Updated markov rmd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jaurbanChicago/jaurbanChicago.github.io/af0f01656774ff3715a103cd0abf5d36d1e09251/docs/Genetic_Drift_Markov.html" target="_blank">af0f016</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-17
</td>
<td>
Added Markov html
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jaurbanChicago/jaurbanChicago.github.io/a311740d4043ff7211b689ab46c8180133bbe94b/docs/Genetic_Drift_Markov.html" target="_blank">a311740</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-17
</td>
<td>
Removed all htmls
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jaurbanChicago/jaurbanChicago.github.io/blob/0d8c343b4a9da2b48bf76780b3460ddd808a7d9e/bin/Genetic_Drift_Markov.Rmd" target="_blank">0d8c343</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-17
</td>
<td>
Updates both rmd and html
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jaurbanChicago/jaurbanChicago.github.io/0d8c343b4a9da2b48bf76780b3460ddd808a7d9e/docs/Genetic_Drift_Markov.html" target="_blank">0d8c343</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-17
</td>
<td>
Updates both rmd and html
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jaurbanChicago/jaurbanChicago.github.io/112890cb54d3e50ade8b37ac80fa7c8c5a5d52ea/docs/Genetic_Drift_Markov.html" target="_blank">112890c</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-17
</td>
<td>
Added Genetic_Drift_Markov.html
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jaurbanChicago/jaurbanChicago.github.io/blob/6b81ff14c4a9b058b7bc47f2b6e8ac4623abde05/bin/Genetic_Drift_Markov.Rmd" target="_blank">6b81ff1</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-17
</td>
<td>
Added Genetic_Drift_Markov.Rmd
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="pre-requisites" class="section level1">
<h1>Pre-Requisites</h1>
<ul>
<li><a
href="https://stephens999.github.io/fiveMinuteStats/wright_fisher_model.html">Wright-Fisher
Model</a></li>
<li><a
href="http://stephens999.github.io/fiveMinuteStats/markov_chains_discrete_intro.html">Discrete
Markov Chains</a></li>
<li>Basic notions of probability theory</li>
<li>Basic notions of population genetics</li>
</ul>
</div>
<div id="introduction-to-discrete-markov-chains" class="section level1">
<h1>Introduction to Discrete Markov Chains</h1>
<p>Before getting into the specific details of the relationship between
Markov chains and genetic drift, it is necessary to briefly explain what
a discrete, finite Markov chain is.</p>
<p>Let´s consider a discrete random variable <span
class="math inline">\(X\)</span>, which at time points <span
class="math inline">\(0,1,2,3,...\)</span>, takes values <span
class="math inline">\(0,1,2,...,M\)</span>. Therefore, it is precise to
say that each state <span class="math inline">\(E_i\)</span>, <span
class="math inline">\(X\)</span> takes the value <span
class="math inline">\(i\)</span> [2]. This means that at each state
<span class="math inline">\(E_i\)</span> at some time <span
class="math inline">\(t\)</span>, <span class="math inline">\(X\)</span>
is in some state <span class="math inline">\(E_i\)</span>. A variable is
said to be <strong>Markovian</strong> if the probability of a certain
outcome in the next time step <span class="math inline">\(t+1\)</span>
depends only on the present state at time <span
class="math inline">\(t\)</span> and is memoryless with regard to any
previous time step <span class="math inline">\(t-1,t-2,...\)</span>.
Therefore, we can now define <strong>a discrete Markov chain</strong> as
a sequence of discrete random variables in which the probability
distribution of the different states at each time <span
class="math inline">\(t\)</span> depends only on the state at time <span
class="math inline">\(t-1\)</span> [1,2].</p>
<p>Here are some important notions on a discrete-time Markov chain
(DTMC) [2,3]. A DTMC is said to be <strong>Markovian</strong> if: <span
class="math display">\[
P(E_{n+1}|E_{n})=i,E_{n-1}=i_{n-1},...,E_1=i_1,E_0=i_0)=P(E_{n+1}=j|E_n=i)\\
\]</span> A DTMC is said to be <strong>time homogeneous</strong> if:
<span class="math display">\[
P(E_{n+1}|E_{n})=P(E_i=j|E_{i-1}=i) \space \mathbb{for \space all}
\space n, \mathbb{so} \space P(E_i=j|E_{i-1})=P_{ij}\\
P_{ij} \gt 0;i,j \ge 0; \sum_{j=0}^{\infty}P_{ij}=1 \\
\]</span> Finally, we can describe the randomness of transitioning
between states using a <strong>transition probability matrix</strong>:
<span class="math display">\[
P_{ij}= \begin{pmatrix}
p_{00} &amp; p_{01} &amp; ... &amp; p_{0M}\\
p_{00} &amp; p_{11} &amp; ... &amp; p_{1M} \\
.\\
.\\
.\\
p_{M0} &amp; p_{M1} &amp; ... &amp; p_{MM}
\end{pmatrix} \\
\]</span> Because the transition probability matrix is a stochastic
matrix, all rows must sum to 1.<br />
<br />
The states of a discrete Markov chain are classified into the following
categories [2,3,4]:</p>
<ul>
<li><strong>Recurrent</strong>:State <span
class="math inline">\(E_i\)</span> can reenter <span
class="math inline">\(E_i\)</span> often infinitely when the chain
starts at this particular <span class="math inline">\(E_i\)</span>.,
i.e. <span class="math inline">\(f_{i}=1\)</span>.</li>
<li><strong>Transient</strong>: State <span
class="math inline">\(E_i\)</span> is transient if, starting at state
<span class="math inline">\(E_i\)</span>, the number of time periods
that the process will be in <span class="math inline">\(E_i\)</span> has
a geometric distribution with mean <span
class="math inline">\(\frac{1}{1-f_i}\)</span>.<br />
</li>
<li><strong>Absorbing</strong>: An state <span
class="math inline">\(E_i\)</span> is absorbinf if the probability of
ever visiting any other state is zero.</li>
</ul>
<p>Next, we are going to classify the chains in two different
classes:</p>
<ul>
<li><strong>Irreducible</strong>: All states can communicate with one
another[3].</li>
<li><strong>Ergodic</strong>: Irreducible, recurrent, and aperiodic.
Aperiodic means all of the Markov chain states are aperiodic [3].</li>
</ul>
<p>For the remainder of this vignette, it is important to denote that we
will be working with Markov chains that have no periodicities in them,
which hardly arise in any genetical applications [2]. Finally, we need
to define the concept of <strong>stationary distribution</strong> of a
Markov chain. A stationary distribution may be defined with the
following mathematical terms [3]:</p>
<p><span class="math display">\[
\text{Consider } P^t \text{ as } t \text{ gets
large}:\lim_{t\to\infty}(\pi^{(0)})^TP^t=\pi^T\\
\text{The } \pi \text{ vector is the limiting distribution of the Markov
chain with stationary property: } \pi^T=\pi^TP \space \mathbb{because}
\space \pi_i=\sum_k \pi_kP_{ki}
\]</span> For ergodic chains, the limiting distribution is also the
stationary distribution and there is only one unique stationary
distribution (i.e., equilibrium distribution) [3]. To obtain the
stationary distribution of the Markov chain, we can use matrix
exponentiation (each row of the resulting matrix will have <span
class="math inline">\(\pi\)</span> in each row), solve linear equations
(solve for <span class="math inline">\(\pi^T=\pi^TP\)</span>), and use
eigendecomposition (<span class="math inline">\(\pi\)</span> is left
eigenvector of <span class="math inline">\(P\)</span>, so
eigendecomposition can lead to <span
class="math inline">\(\pi\)</span>).</p>
</div>
<div id="markov-chains-in-genetic-drift" class="section level1">
<h1>Markov Chains in Genetic Drift</h1>
<p>A <strong>discrete Markov chain</strong> can be quite helpful to
describe genetic drift in an ensemble of numerous identical and
independent populations of a Wright-Fisher model [1]. A Markov chain
model will help us understand how drift modifies, on average, the
alleles frequencies of a biallelic locus in various population
replicates through time. The <strong>transition probability</strong> for
the Markov chain describing allele frequencies changes in a specific
population (from an ensemble of replicate populations) is [3]:</p>
<p><span class="math display">\[
P_{ij}=P(p_{t+1}=\frac{j}{2N}|p_t=\frac{i}{2N})={2N \choose
j}(\frac{i}{2N})^j(1-\frac{i}{2N})^{2N-j} \\
\mathbb{E}[p_{t+1}=\frac{j}{2N}|p_t=\frac{i}{2N}]=p_t\\
Var(p_{t+1}=\frac{j}{2N}|p_t=\frac{i}{2N})=\frac{p_t(1-p_t)}{2N}
\]</span> This transition probability equation describes the probability
that a given Wright-Fisher population with a exactly <span
class="math inline">\(i\)</span> copies of allele <span
class="math inline">\(A\)</span> in generation t has exactly <span
class="math inline">\(j\)</span> copies of <span
class="math inline">\(A\)</span> in generation <span
class="math inline">\(t+1\)</span>. This equation is another
confirmation that the allele changes between discrete time units is
<strong>Markovian</strong>, which means that the allele trajectories in
<span class="math inline">\(t+1\)</span> will exclusively depend on the
frequencies at <span class="math inline">\(t\)</span>[1-4]. A Markov
chain describing genetic drift (no mutation and no selection) can have
two <strong>absorbing states</strong>, which are when the allele
frequency <span class="math inline">\(p_i=0\)</span> (loss) or <span
class="math inline">\(p_i=1\)</span> (fixation) [5]. Besides these two
absobing states, allele frequency can, in principle, change from any
state to other states in a single generation. However, how big or small
is the change between generations, will depend, in principle, in <span
class="math inline">\(N\)</span> and the frequency itself (remember the
binomial sampling from the introductory <a
href="https://jaurbanchicago.github.io/docs/Genetic_Drift_Intro.html">vignette</a>).</p>
</div>
<div id="code-simulations-of-genetic-drifts-using-markov-chains"
class="section level1">
<h1>Code Simulations of Genetic Drifts using Markov Chains</h1>
<p>We will use some simple code to simulate and model random genetic
drift as a simple discrete Markov chain.</p>
<pre class="r"><code># We are going to give some examples of random genetic drift trajectories using the transition probability (Pij). 
#We are going to use the CRAN R package learnPopGen created by Liam J. Revell [5]

library(learnPopGen)

# Initial frequency p=0.01,N=50,500 and 50 population replicates

genetic.drift(p0=0.01,Ne=50,nrep=50, time=100, show = &quot;p&quot;,pause = 0.1)</code></pre>
<p><img src="figure/Genetic_Drift_Markov.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-1">
Past versions of unnamed-chunk-1-1.png
</button>
</p>
<div id="fig-unnamed-chunk-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jaurbanChicago/jaurbanChicago.github.io/blob/dcdda13025f5bc6fad0606f0b1eafdb20e89c99f/docs/figure/Genetic_Drift_Markov.Rmd/unnamed-chunk-1-1.png" target="_blank">dcdda13</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>genetic.drift(p0=0.01,Ne=500,nrep=50, time=100, show = &quot;p&quot;,pause = 0.1)</code></pre>
<p><img src="figure/Genetic_Drift_Markov.Rmd/unnamed-chunk-1-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-2">
Past versions of unnamed-chunk-1-2.png
</button>
</p>
<div id="fig-unnamed-chunk-1-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jaurbanChicago/jaurbanChicago.github.io/blob/dcdda13025f5bc6fad0606f0b1eafdb20e89c99f/docs/figure/Genetic_Drift_Markov.Rmd/unnamed-chunk-1-2.png" target="_blank">dcdda13</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Initial frequency p=0.5,N=50,500 and 50 population replicates

genetic.drift(p0=0.5,Ne=50,nrep=50, time=100, show = &quot;p&quot;,pause = 0.1)</code></pre>
<p><img src="figure/Genetic_Drift_Markov.Rmd/unnamed-chunk-1-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-3">
Past versions of unnamed-chunk-1-3.png
</button>
</p>
<div id="fig-unnamed-chunk-1-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jaurbanChicago/jaurbanChicago.github.io/blob/dcdda13025f5bc6fad0606f0b1eafdb20e89c99f/docs/figure/Genetic_Drift_Markov.Rmd/unnamed-chunk-1-3.png" target="_blank">dcdda13</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>genetic.drift(p0=0.5,Ne=500,nrep=50, time=100, show = &quot;p&quot;,pause = 0.1)</code></pre>
<p><img src="figure/Genetic_Drift_Markov.Rmd/unnamed-chunk-1-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-4">
Past versions of unnamed-chunk-1-4.png
</button>
</p>
<div id="fig-unnamed-chunk-1-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jaurbanChicago/jaurbanChicago.github.io/blob/dcdda13025f5bc6fad0606f0b1eafdb20e89c99f/docs/figure/Genetic_Drift_Markov.Rmd/unnamed-chunk-1-4.png" target="_blank">dcdda13</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Initial frequency p=0.8,N=50,500 and 50 population replicates

genetic.drift(p0=0.8,Ne=50,nrep=50, time=100, show = &quot;p&quot;,pause = 0.1)</code></pre>
<p><img src="figure/Genetic_Drift_Markov.Rmd/unnamed-chunk-1-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-5">
Past versions of unnamed-chunk-1-5.png
</button>
</p>
<div id="fig-unnamed-chunk-1-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jaurbanChicago/jaurbanChicago.github.io/blob/dcdda13025f5bc6fad0606f0b1eafdb20e89c99f/docs/figure/Genetic_Drift_Markov.Rmd/unnamed-chunk-1-5.png" target="_blank">dcdda13</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>genetic.drift(p0=0.8,Ne=500,nrep=50, time=100, show = &quot;p&quot;,pause = 0.1)</code></pre>
<p><img src="figure/Genetic_Drift_Markov.Rmd/unnamed-chunk-1-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-6">
Past versions of unnamed-chunk-1-6.png
</button>
</p>
<div id="fig-unnamed-chunk-1-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jaurbanChicago/jaurbanChicago.github.io/blob/dcdda13025f5bc6fad0606f0b1eafdb20e89c99f/docs/figure/Genetic_Drift_Markov.Rmd/unnamed-chunk-1-6.png" target="_blank">dcdda13</a>
</td>
<td>
jaurbanChicago
</td>
<td>
2022-03-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>After taking a look at these simulations, we can observe that the
size of <span class="math inline">\(N\)</span> indeed affects the
variability of the outcome of the random sampling in the Markov chain.
We can also see that once <span class="math inline">\(A\)</span> reaches
fixation or gets lost in one of the replicates, it can never get out of
that particular <strong>absorbing state</strong>. Finally, the frequency
in each state <span class="math inline">\(t\)</span> affects the
variability of the trajectories and this becomes even more apparent when
the allele <span class="math inline">\(A\)</span> is either close to
fixation or loss in each replicate.</p>
</div>
<div
id="intractability-of-genetic-drift-as-a-markov-chain-in-a-wright-fisher-model"
class="section level1">
<h1>Intractability of Genetic Drift as a Markov chain in a Wright-Fisher
Model</h1>
<p>One of the caveats of modeling genetic drift as a Markov chain in a
Wright-Fisher population is the fact that it becomes mathematically
intractable. This means that it is very difficult to obtain simple
analytical results on the distributions of allele frequencies after
<span class="math inline">\(t\)</span> generations of genetic drift [6].
We can iterate over and over the Markov chain transition probabilities
and we would never derive analytical formulas to describe the allele
frequencies distributions. <a
href="https://jaurbanchicago.github.io/docs/diffusion_process.html">Diffusion
theory</a> has been used to approximate certain features of the
distribution (<strong>it´s still not easy and will probably require some
numerical techniques!!!</strong>) and other interesting results
(fixation probabilities,equilibrium allele frequency spectrum, etc) [6].
You can find more about diffusion approximations of genetic drift in the
following <a
href="https://jaurbanchicago.github.io/docs/diffusion_approx.html">vignette</a>.</p>
</div>
<div id="references" class="section level1">
<h1>References</h1>
<ol style="list-style-type: decimal">
<li>Hamilton, M. B. (2008). <em>Population genetics</em>. John Wiley
&amp; Sons, 53-73.<br />
</li>
<li>Ewens, W. J. (2004). Mathematical population genetics: theoretical
introduction (Vol. 1). New York: Springer, 86-91.<br />
</li>
<li>Hartl, D. L. (2020).<em>A primer of population genetics and
genomics</em>. Oxford University Press,147-173.</li>
<li>Novembre,J. “Lecture: Discrete Markov Chains”,February 2022,
University of Chicago, Chicago, IL. 5.Revell, L. J. (2019). learnPopGen:
An R package for population genetic simulation and numerical analysis.
Ecology and Evolution, 9(14), 7896-7902.</li>
<li>Berg,J.J. “Lecture:Forward Models 1”,February 2022, University of
Chicago, Chicago, IL.</li>
</ol>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.2 (2021-11-01)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] learnPopGen_1.0.4

loaded via a namespace (and not attached):
 [1] phangorn_2.8.1          gtools_3.9.2            xfun_0.29              
 [4] bslib_0.3.1             lattice_0.20-45         phytools_1.0-1         
 [7] vctrs_0.3.8             expm_0.999-6            htmltools_0.5.2        
[10] yaml_2.2.2              utf8_1.2.2              rlang_1.0.2            
[13] jquerylib_0.1.4         later_1.3.0             pillar_1.7.0           
[16] glue_1.6.2              lifecycle_1.0.1         stringr_1.4.0          
[19] combinat_0.0-8          workflowr_1.7.0         codetools_0.2-18       
[22] coda_0.19-4             evaluate_0.14           knitr_1.37             
[25] fastmap_1.1.0           httpuv_1.6.5            parallel_4.1.2         
[28] fansi_1.0.2             highr_0.9               Rcpp_1.0.8.2           
[31] promises_1.2.0.1        plotrix_3.8-2           clusterGeneration_1.3.7
[34] scatterplot3d_0.3-41    jsonlite_1.8.0          tmvnsim_1.0-2          
[37] fs_1.5.2                fastmatch_1.1-3         mnormt_2.0.2           
[40] digest_0.6.29           stringi_1.7.6           numDeriv_2016.8-1.1    
[43] rprojroot_2.0.2         grid_4.1.2              quadprog_1.5-8         
[46] cli_3.2.0               tools_4.1.2             magrittr_2.0.2         
[49] sass_0.4.0              maps_3.4.0              tibble_3.1.6           
[52] crayon_1.5.0            ape_5.6-2               whisker_0.4            
[55] pkgconfig_2.0.3         ellipsis_0.3.2          MASS_7.3-55            
[58] Matrix_1.4-0            rmarkdown_2.11          rstudioapi_0.13        
[61] R6_2.5.1                igraph_1.2.11           nlme_3.1-155           
[64] git2r_0.29.0            compiler_4.1.2         </code></pre>
</div>
</div>

<hr>
<p>
    This site was created with <a href="http://rmarkdown.rstudio.com">R Markdown</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
